src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -434,14 +434,24 @@ private SafeCurlHandle CreateRequestHandle(RequestCompletionSource state, GCHand
                 // Set maximum automatic redirection option
                 SetCurlOption(requestHandle, CURLoption.CURLOPT_MAXREDIRS, _maxAutomaticRedirections);
             }
-            if (state.RequestMessage.Content != null)
+
+            if (state.RequestMessage.Method == HttpMethod.Put)
```

Comment:
PUT or POST without a body is valid (and equivalent to sending a zero-octet message) though people have coded to the assumption that it's not valid in the past. On the one hand this could mean null content with zero-length content would be safer, but on the other hand there could well be a server out there that (incorrectly) treats no body as allowed by different to zero-octet body, and such a "safety" mechanism would prevent working to be bug-compatible with that server.